From 3dabcc8463066b5cf5ac1b1eff3f5c3f04784d9c Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Wed, 28 Jun 2023 00:25:38 +0530
Subject: [PATCH] Fix crash seen during unloading of oneVPL GPU runtime

While VPL dispatch tries to unload all libaries, crash seen
in oneVPL GPU runtime release.

As the PerfUtility member variables are not initialized properly,
usage of macro PERF_UTILITY_PRINT tries to create folder which
results in crash.

Fix the issue by initializing member variables of PerfUtility to
0 that will make sure perf logging is disabled.

Change-Id: I146a4907bfa5658230ebb42846dc0651fba31aa4
Tracked-On: OAM-110789
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 _studio/shared/mfx_logging/src/mfx_utils_perf.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/_studio/shared/mfx_logging/src/mfx_utils_perf.cpp b/_studio/shared/mfx_logging/src/mfx_utils_perf.cpp
index 2f4cf642..657c4a28 100644
--- a/_studio/shared/mfx_logging/src/mfx_utils_perf.cpp
+++ b/_studio/shared/mfx_logging/src/mfx_utils_perf.cpp
@@ -85,8 +85,8 @@ PerfUtility* PerfUtility::getInstance()
 
 PerfUtility::PerfUtility()
 {
-    int32_t dwPerfUtilityIsEnabled = 0;
-    double timeStamp = 0;
+    dwPerfUtilityIsEnabled = 0;
+    timeStamp = 0;
 }
 
 PerfUtility::~PerfUtility()
@@ -193,4 +193,4 @@ void PerfUtility::printPerfTimeStamp(Tick *newTick, const std::vector<uint32_t>&
         }
     }
     log_buffer[current_tid].append("\n");
-}
\ No newline at end of file
+}
-- 
2.41.0

